### PR TITLE
Added GetSkipAndCount() for reuse in pagination

### PR DESCRIPTION
**Pull request by jmbrown412**
Added a helper method GetSkipAndCount() to get skip and count params for resuse in pagination in Moov apis.

### CODE REVIEW

File: http/server.go [hunk 1]
```diff
@@ -112,3 +113,17 @@ func GetRequestID(r *http.Request) string {
 func GetUserID(r *http.Request) string {
 	return strx.Or(r.Header.Get("X-User"), r.Header.Get("X-User-Id"))
 }
+
+// ReadSkipAndCount returns the skip and count pagination values from the query
```
Comment (adamdecaf) [inline http/server.go:117]:
```suggestion
// GetSkipAndCount returns the skip and count pagination values from the query
```

Comment (adamdecaf) [reply http/server.go:117]: Can we add a sentence about what "skip" and "count" mean to the UI or a calling client?

File: http/server.go [hunk 2]
```diff
@@ -112,3 +113,17 @@ func GetRequestID(r *http.Request) string {
...
+func GetSkipAndCount(r *http.Request) (skip int, count int, exists bool) {
+	skip, err := strconv.Atoi(r.URL.Query().Get("skip"))
+	if err != nil {
+		skip = 0
+	}
+	count, err = strconv.Atoi(r.URL.Query().Get("count"))
```
Comment (adamdecaf) [inline http/server.go:123]: Can we introduce a hard limit? 200 was what we agreed to in slack. `math.Min(count, 200)`

File: http/server.go [hunk 3]
```diff
@@ -112,3 +113,17 @@ func GetRequestID(r *http.Request) string {
 func GetUserID(r *http.Request) string {
 	return strx.Or(r.Header.Get("X-User"), r.Header.Get("X-User-Id"))
 }
+
+// ReadSkipAndCount returns the skip and count pagination values from the query
+func GetSkipAndCount(r *http.Request) (skip int, count int, exists bool) {
+	skip, err := strconv.Atoi(r.URL.Query().Get("skip"))
```
Comment (adamdecaf) [inline http/server.go:119]: Can we cap this? 10,000?

File: http/server.go [hunk 4]
```diff
@@ -112,3 +113,17 @@ func GetRequestID(r *http.Request) string {
...
+	if err != nil {
+		skip = 0
+	}
+	count, err = strconv.Atoi(r.URL.Query().Get("count"))
+	if err != nil {
+		count = 0
+	}
+	exists = skip > 0 || count > 0
```
Comment (adamdecaf) [inline http/server.go:127]: I was thinking `exists` is only `true` if either skip or count are specified in the URL. We might want to know they're invalid values that got converted to 0.

File: http/server.go [hunk 5]
```diff
@@ -112,3 +113,17 @@ func GetRequestID(r *http.Request) string {
...
+	return skip, count, exists
```
Comment (adamdecaf) [inline http/server.go:128]: Just realized we should prevent negative numbers. `math.Max(0, skip)` (same for count)

**Review by adamdecaf — COMMENTED** [review]
Sorry Jordan for bombing this with a lot of comments.

File: http/server.go (hunk 1 above)
Comment (jmbrown412) [reply http/server.go]: Done

File: http/server.go (hunk 2 above)
Comment (jmbrown412) [reply http/server.go]: Done

File: http/server.go (hunk 3 above)
Comment (jmbrown412) [reply http/server.go]: Done

File: http/server.go (hunk 4 above)
Comment (jmbrown412) [reply http/server.go]: Done

File: http/server.go (hunk 5 above)
Comment (jmbrown412) [reply http/server.go]: Done

File: http/server.go [hunk 6]
```diff
@@ -112,3 +114,31 @@ func GetRequestID(r *http.Request) string {
...
+// count is the number of records to retrieve in the search
+// exists indicates if skip or count was passed into the request URL
+func GetSkipAndCount(r *http.Request) (skip int, count int, exists bool, errors []error) {
+	skipVal := r.URL.Query().Get("skip")
+	var err error
+	skip, err = strconv.Atoi(skipVal)
+	skip = int(math.Min(float64(skip), 10000))
+	skip = int(math.Max(0, float64(skip)))
+	if err != nil && skip == 0 && skipVal != "" {
+		errors = append(errors, err)
```
Comment (adamdecaf) [inline http/server.go:129]: I think we should return the first error. If we can't parse skip/count then we should likely fail the request.

Comment (jmbrown412) [reply http/server.go:127]: I updated it to append the err only if they passed in a value so we won't return errors if they didn't pass in anything.

File: http/server.go [hunk 7]
```diff
@@ -112,3 +114,42 @@ func GetRequestID(r *http.Request) string {
...
+func GetSkipAndCount(r *http.Request) (skip int, count int, exists bool, errors []error) {
+	skipVal := r.URL.Query().Get("skip")
+	var err error
+	skip, err = strconv.Atoi(skipVal)
+	if err != nil && len(skipVal) > 0 {
+		errors = append(errors, err)
+		skip = 0
+	}
+	// TODO - Add this to a config somewhere?
+	// Cap skip at 10000
```
Comment (adamdecaf) [inline http/server.go:131]: I don't think we care about these being configurable. They're going to be hardcoded across services.

Comment (jmbrown412) [reply http/server.go:130]: Makes sense. Pushed.

File: http/server.go (hunk 6 above)
Comment (adamdecaf) [reply http/server.go:127]: I don't understand why we're returning `[]error` though. Can we just return the first `error`?

Comment (jmbrown412) [reply http/server.go]: Yeah, that was confusing. I was trying to be too informative. Refactored to just return an err.

**Review by adamdecaf — APPROVED** [review]

**Pull request merged**